src/main/java/com/marklogic/mule/extension/Operations.java

```diff
@@ -165,8 +169,18 @@ public List<Result<InputStream, DocumentAttributes>> getPage(DatabaseClient data
                     queryDefinition.setResponseTransform(serverTransform);
                 }
 
+                if ((useConsistentTimestamp) && (searchTimeStamp == -1)) {
+                    JacksonParserHandle searchHandle = new JacksonParserHandle();
+                    documentManager.setPageLength(1);
+                    documentManager.search(queryDefinition,1, searchHandle);
```

Comment:
A mild optimization here would be that if we don't get any results here, we don't need the follow up query. Try it out and if all the tests pass (and you already added "no match" tests which is good), I think it's a good and very simple optimization to have.